File: Data Structures & Algorithms/climbing-stairs/submission-5.py
class Solution:
    def climbStairs(self, n: int) -> int:
        
        one, two = 0, 1

        for i in range(n):
            tmp = two
            two = two + one
            one = tmp
        return two

        # climbStairs(n - 1) + climbStairs(n - 2)

# 1 2 3 4 5 
# 5 3 2 1 1
#       o t

File: Data Structures & Algorithms/house-robber/submission-5.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int one = 0, two = nums[0];

        for (int i = 1; i < nums.size(); i++) {
            int tmp = two;
            two = max(one + nums[i], two);
            one = tmp;
        }
        return two;
    }
};
